examples/serverless/publish.js

```diff
@@ -1,16 +1,30 @@
+if (process.env.CI !== "true") {
+  console.log("skipping Pact publish as not on CI...")
+  process.exit(0)
+}
+
 const pact = require("@pact-foundation/pact-node")
-const path = require("path")
+const childProcess = require("child_process")
+
+const exec = command =>
+  childProcess
+    .execSync(command)
+    .toString()
+    .trim()
+
+//Usually, you would just use the CI env vars, but to allow these examples to run from
+//local development machines, we'll fall back to the git command when the env vars aren't set.
+const gitSha = process.env.TRAVIS_COMMIT || exec("git rev-parse HEAD")
+const branch =
+  process.env.TRAVIS_BRANCH || exec("git rev-parse --abbrev-ref HEAD")
+
 const opts = {
-  pactFilesOrDirs: [path.resolve(__dirname, "pacts/")],
+  pactFilesOrDirs: ["./pacts/"],
```

Comment:
Unless there's a subtlety I am missing, I think the `__dirname` version was more robust, as it is defined to be the same place as this script.

Reply:
That's correct Tim. If somebody runs the script from a different directory it will fail.